src/Appwrite/Platform/Workers/Builds.php

```diff
@@ -210,6 +194,12 @@ protected function buildDeployment(callable $getFunctionsDevice, Func $queueForF
                 $stdout = '';
                 $stderr = '';
                 Console::execute('mkdir -p /tmp/builds/' . \escapeshellcmd($buildId), '', $stdout, $stderr);
+
+                $build = $dbForProject->getDocument('builds', $buildId);
+                if ($build->getAttribute('status') === 'cancelled') {
+                    return;
+                }
```

Comment:
I see this check above, do we need to duplicate it here?

Reply:
I see duplication in other places too

Reply:
The duplication is intentional, since cancellation can happen at any stage and we should save compute by capturing it before time consuming tasks.